### ISSUE TITLE
Add product ID and icon url to csv export

### ISSUE THREAD

**Issue by rvdsoft** [RePod/psdle#30]
The icon is shown in the export view but when you export to csv, there is no icon field with an icon url. 
Also you can filter using the product ID but the field is not shown. So it would be very much appreciated if you could add the product ID and icon url to the csv export.

EDIT: Oh yes sorry, I meant that the icon is shown in the list but not in the export view.

**Comment by RePod** [comment]
![screenshot_1](https://cloud.githubusercontent.com/assets/7897871/14035011/891409d0-f1ff-11e5-9559-bd0ba2bf520c.jpg)

You're referring to this? I apologize, your original wording suddenly confused me that I allowed the icon URL to be exported and had no recollection of it.

This is similar to #26, and I have an idea to rewrite the exporting options to export whatever you want (whatever entry info PSDLE stores) but I'm lazy.

It will happen and satisfy everybody, but I don't have a time frame for it.
More so because there's one hurdle I don't know how to overcome without making it really slow.

**Comment by RePod** [comment]
![screenshot_2](https://cloud.githubusercontent.com/assets/7897871/14037563/ecaeccca-f21b-11e5-9370-b7d7e7d58ad0.jpg)

Here's how the new export window looks (it's rough, but functional).
You can name columns whatever you want and choose what property they represent.
You can add or subtract as many columns as desired, *literally*.

Please try it out (bug test if you like) by doing the following:
 - Open your browser console:
   - Firefox: `Ctrl + Shift + K`
   - Chrome: `Ctrl + Shift + J` 
 - Paste [the contents of this link into it](https://raw.githubusercontent.com/RePod/psdle/a9134ebe179fafb991e94096bba86654a39d3b6e/psdle.js) and submit.
 - You may close the browser console with the same key combination.

**Comment by rvdsoft** [comment]
I've tested the new version and for the columns that matter it finds the correct data. Great job!
One problem: If you add a field in the export view, the "Column Name" is empty by default. Can't you make it so that when you select from the fields combo box, the column name changes as well instead of staying empty?

I have a game tracker app and I want to give users the option to import from a PSDLE csv file. But I need to know the column names beforehand in order do that properly.

**Comment by Tyrael87** [comment]
Thanks for improving the exporting options. I wished for Product ID export, too. Now i have a couple of questions...

1) Is it possible to save the settings of the Column Name and Property somewhere, so you don't have to choose/name them every time?
2) I've noticed that you can also export the Category of each item now. But do you think it's possible to remove all those "unknown" entries if we know exactly the actual category? I noticed that some themes still have "unknown" as the category, or some things that are not themes are labeled as such (i.e. some Darkstalkers Resurrection DLC)... i had to search/count for them manually but it would be great if at least all the themes and avatars appear in their proper category.

**Comment by RePod** [comment]
>One problem: If you add a field in the export view, the "Column Name" is empty by default. Can't you make it so that when you select from the fields combo box, the column name changes as well instead of staying empty?

Not possible while also maintaining translations, unless you mean literally copying over whatever is selected without translating it. End users would still be able to change them from their defaults.

At most, I plan to only apply the actual translations PSDLE already has (the columns you see on the table).

**Comment by RePod** [comment]
For the previous user who commented with suggestions, I got them via email so I'll still address them:

>Is it possible to save the settings of the Column Name and Property somewhere, so you don't have to choose/name them every time?

Export settings are stored temporarily in the current session as long as you export as normal.
Closing the window without exporting will discard the current settings.

As mentioned in the previous comment I could potentially include the settings used in the export itself then add an import function (which would get kind of crazy).

>But do you think it's possible to remove all those "unknown" entries if we know exactly the actual category? I noticed that some themes still have "unknown" as the category, or some things that are not themes are labeled as such

That's at discretion of the Catalog, unless `category` is available without Catalog enabled thus all are `unknown`. The best I'd be able to do is make it empty if so.

**Comment by RePod** [comment]
````
PSDLE_EXPORT_SETTINGS	[{"name":"Name","target":"name"},{"name":"Platform","target":"platform"},{"name":"Size","target":"size_pretty"},{"name":"Date","target":"date_pretty"}]
````

d9ae939ebb61735ea6ff7acbff367e582bf3c63a The footer now includes the settings used, split with the provided separator.
This isn't pushed to live though, yet.

**Comment by Tyrael87** [comment]
So settings saving is coming in the near future? Good. Anyway, Catalog is never enabled by default, at least for me. But i think it was in the past. I don't remember when/if it changed. Huh.

And no worries about the "unknown" category, i'll live with it. Thank goodness i manually created a list of avatars/themes in the past... it's hilarious that the Darkstalkers cabinet themes are treated like actual PS3 themes.

ps: is the "item counter starting at a certain number (400-something for me) and then decreasing and again increasing normally" bug still supposed to happen as of today?

**Comment by RePod** [comment]
>So settings saving is coming in the near future? Good.
In a sense. I don't want PSDLE to actually store anything (even settings) permanently, just in the current session..

I'll probably add an import thing to the export window you can paste a pre-made line similar to the one above and get similar results, but that's it.

>is the "item counter starting at a certain number (400-something for me) and then decreasing and again increasing normally" bug still supposed to happen as of today?

That's not a bug, that's just some additional Catalog requests for things that share product IDs. When the bar actually starts going up it's fetching the individual/unique item IDs (that aren't the product IDs). 

Basically, Catalog is two steps. It just so happens the bar decreases to indicate the first step.

**Comment by Tyrael87** [comment]
Awww, too bad.

ps: there's no way to see/export PSTV compatibility anymore? (with US accounts)

**Comment by rvdsoft** [comment]
> Not possible while also maintaining translations, unless you mean literally copying over whatever is selected without translating it. End users would still be able to change them from their defaults.
> 
> At most, I plan to only apply the actual translations PSDLE already has (the columns you see on the table).
> 
> I could also try including the export settings as JSON with the export info, similar to what PSDLE uses to generate the export. It'd be something like [{name:'Date', target:'date_pretty'}, ...}]. Done, see two comments below.

I've decided to just find out the columns based on the values. 

One other problem: 
When you select the description field it contains newline characters which ruins the whole csv file.
So all newlines should be removed.

**Comment by RePod** [comment]
>ps: there's no way to see/export Vita/PSTV compatibility anymore? (with non-EU accounts)

These are now available again as the `vitaCompat` and `vitatvCompat` properties.

>When you select the description field it contains newline characters which ruins the whole csv file.
So all newlines should be removed.

I'd question why people would want to export the description, but that's why the new export exists.

Both should be addressed, but again they're not on live.

**Comment by Tyrael87** [comment]
@RePod really? I'm not seeing them at all...

**Comment by RePod** [comment]
The changes haven't been made live/confirmed stable yet, but the files available on this repository have them available.

They should be good to go in a few days at the most then you'll see them across all versions of PSDLE.

**Comment by Tyrael87** [comment]
Ok, thank you.

**Comment by Tyrael87** [comment]
Ok, thank you.

ps: i guess it's a temporary issue that i get the categories at the bottom of the exported list?

**Comment by RePod** [comment]
>i guess it's a temporary issue that i get the categories at the bottom of the exported list?

Are you referring to the versions in this repository? If so, that's not temporary (unless I feel like reverting it in the future). As it is I currently have the last line of the export indicate what property the column is with a new column for the complete JSON:

| Name | Date | |
| --- | --- | --- |
| Game | 01/01/2016 | |
| name | prettyDate | [{'name': 'Name', 'target': 'name'}, {'name': 'Date', 'target': 'prettyDate'}]


>And the only PSTV-compatible entry i get is Need for Speed Most Wanted for Vita? I also have a couple of PSOne Classics and at least one or two of them should be PSTV-compatible according to several lists.

If I recall, your store was it-it. PS TV support shouldn't even be available unless you're using an en-us account/store. The reason it's tied to en-us is because it looks up a pre-determined URL in the Catalog which contains all the PS TV supported games, and I haven't looked into other regions yet.

**Comment by Tyrael87** [comment]
No RePod, in that case i was talking about the US store. I know very well that the PSTV doesn't even appear with EU (or JP) accounts. I have three different PSN accounts.

**Comment by RePod** [comment]
You're still at the mercy of what the store lists as compatible.

![screenshot_3](https://cloud.githubusercontent.com/assets/7897871/14062809/534c8252-f383-11e5-875c-6848fa6ce5e2.jpg)

As shown here, only the second game is marked as PS TV compatible. However, the first is actually incompatible with PS TV and the third is but not marked as such on the store.

**Comment by RePod** [comment]
I guess the original scope of this issue has been resolved by the new export window.

At the time of writing this, the latest commit added a new `baseGame` property if Catalog is enabled and the info needed is available. Without any additional changes, the export window was instantly capable of exporting it. Therefore, in the future all, except problematic, properties should be available with no hassle.
